From 879bcccf22319886a1c428d82c773fa7f78f8548 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Fri, 9 Mar 2018 15:37:43 +0700
Subject: [PATCH 402/909] iommu/ipmmu-vmsa: Invalidate TLB when IPMMU
 translation error occurred

commit 70b682f24e008112eeee10228cb19b4d998025ac from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Invalidate TLB when IPMMU translation error occurred
as described in hardware manual.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iommu/ipmmu-vmsa.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index 02a52ecc19cb..9717a048f3be 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -691,6 +691,9 @@ static irqreturn_t ipmmu_domain_irq(struct ipmmu_vmsa_domain *domain)
 	if (!(status & (IMSTR_PF | IMSTR_TF)))
 		return IRQ_NONE;
 
+	/* Flush the TLB as required when IPMMU translation error occurred. */
+	ipmmu_tlb_invalidate(domain);
+
 	/*
 	 * Try to handle page faults and translation faults.
 	 *
-- 
2.17.1

